Features/Context/SubContext/CommonActions.php

```diff
@@ -287,11 +315,22 @@ public function selectFromUniversalDiscovery($path)
 
     /**
      * @When I confirm the selection
-     * Confirm selection in Universal descovery.
+     * Confirm selection in Universal discovery.
      */
     public function confirmSelection()
     {
-        $this->clickElementByText('Confirm selection', '.ez-universaldiscovery-confirm');
+        $elem = $this->findWithWait('.ez-view-universaldiscoveryview');
+        $elem->find('css', '.ez-universaldiscovery-confirm')->click();
+    }
+
+    /**
+     * @When I cancel the selection
+     * Cancel the selection in Universal discovery.
+     */
+    public function cancelSelection()
```

Comment:
A bit bothered by the fact that UDW sentences and methods are part of `CommonActions`. Could you try to change that ?